apexlog/apexlog.go

```diff
@@ -0,0 +1,96 @@
+package apexlog
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/airbrake/gobrake/v5"
+
+	"github.com/apex/log"
+)
+
+// severity identifies the sort of log: info, warn etc.
+type severity int
+
+// These constants identify the log levels in order of increasing severity.
+// A message written to a high-severity log file is also written to each
+// lower-severity log file.
+const (
+	debugLog severity = iota
+	infoLog
+	warnLog
+	errorLog
+	fatalLog
+)
+
+var severityName = []string{
+	debugLog: "debug",
+	infoLog:  "info",
+	warnLog:  "warn",
+	errorLog: "error",
+	fatalLog: "fatal",
+}
+
+// String implementation for severity.
+func (s severity) String() string {
+	return severityName[s]
+}
+
+func parseSeverity(s string) (severity, bool) {
```

Comment:
We don't need this function if `severity` is of type `severity`, not `string`.

Reply:
I didn't wanted to expose the severity variable in the project, it was to make sure no one can modify it or may lead to memory leakage issue. It needs to be inside the package, if you want we can define constants that can be used outside the package.

Reply:
On line 18 you can export constants outside the package. Then the users will be able to pass severity like this:

```
apexlog.New(notifier, apexlog.DebugLog)
```

I don't see where we can leak memory. It's basically how the `log` package works. In this PR we have:

```
log.SetLevel(log.DebugLevel)
```

That's the same as `apexlog.New(notifier, apexlog.DebugLog)`